### PR TITLE
Implement rack middleware

### LINKED ISSUE #13: Implement RackMiddleware
RackMiddleware has a placeholder implementation and needs to be implemented.

### PR DESCRIPTION
**Pull request by chingor13**
Fixes #13

### CODE REVIEW

File: lib/opencensus/trace/integrations/rack_middleware.rb [hunk 1]
```diff
@@ -15,24 +15,86 @@
...
+          span.put_attribute "/http/client_protocol", env["SERVER_PROTOCOL"]
+          span.put_attribute "/http/user_agent", env["HTTP_USER_AGENT"]
+          span.put_attribute "/pid", ::Process.pid.to_s
+          span.put_attribute "/tid", ::Thread.current.object_id.to_s
+        end
+
+        def finish_request span, response
+          if response.is_a?(::Array) && response.size == 3
+            span.set_status response[0]
+            response[1]["Trace-Context"] = span.context.to_trace_context_header
```
Comment (chingor13) [inline lib/opencensus/trace/integrations/rack_middleware.rb]: Also see #29

Comment (dazuma) [reply lib/opencensus/trace/integrations/rack_middleware.rb]: This is a response header. Is Trace-Context defined on responses?

Comment (chingor13) [reply lib/opencensus/trace/integrations/rack_middleware.rb]: I guess it's not. Propagating the `X-Cloud-Trace` header back upstream was part of the Stackdriver Trace header requirements. I can remove it here.

Comment (chingor13) [reply lib/opencensus/trace/integrations/rack_middleware.rb]: Although I do remember conversations about returning the header in the response.

Comment (dazuma) [reply lib/opencensus/trace/integrations/rack_middleware.rb]: I know there have been conversations, but I haven't seen a solid decision or spec on the semantics of the response header. We can check the other language implementations to see if anyone is actually setting a header.

Comment (chingor13) [reply lib/opencensus/trace/integrations/rack_middleware.rb]: Ok, I don't see anyone else except php that's doing it. I'll remove it here.

**Pull request merged**